src/parser/mod.rs

```diff
@@ -14578,4 +14584,26 @@ mod tests {
 
         assert!(Parser::parse_sql(&MySqlDialect {}, sql).is_err());
     }
+
+    #[test]
+    fn test_keyword() {
+        let sql = "select * from t where user = 'test';";
```

Comment:
oh actually the problem being solved isn't currently clear to me, my understanding is that in postgres for example, given this statement, `user` is indeed a function. What's the undesirable behavior being solved for by this PR?